### PR TITLE
Get rid of PixelAccessor<T>

### PR DESCRIPTION
**Pull request by antonfirsov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
`PixelAccessor<T>` has been originally used to allow efficient pixel manipulation through pinning + pointer arithmetics. Thanks to `Unsafe.Add()` and `Span<T>` pinning is no longer necessary, but we kept `PixelAccessor<T>` as a legacy component (mostly used in tests) to minimize refactor efforts.

- This PR replaces all `PixelAccessor<T>` usages with `Buffer2D<T>` and removes the class from the codebase. 
- The `IBuffer2D<T>` abstraction is no longer necessary, so it has been replaced with `Buffer2D<T>`.

I think the changes are trivial, so I plan to merge it as soon as I get at least one quick review.

### CODE REVIEW

File: src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs [hunk 1]
```diff
@@ -574,6 +580,7 @@ private void ReadImageHeaders(Stream stream, out bool inverted, out byte[] palet
                 this.stream.Read(palette, 0, colorMapSize);
             }
 
+            // TODO: ReSharper tells this expression is always true, looks like he's pretty right about it:
             if (this.infoHeader.Width > int.MaxValue || this.infoHeader.Height > int.MaxValue)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs]: You mean always false yeah? Been struggling to find a definitive answer but I think 65535x65535 is the maximum size on a 64 bit machine.

https://stackoverflow.com/questions/29175585/what-is-the-maximum-resolution-of-c-sharp-net-bitmap

File: src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs [hunk 2]
```diff
@@ -103,36 +103,42 @@ public Image<TPixel> Decode<TPixel>(Stream stream)
                 this.ReadImageHeaders(stream, out bool inverted, out byte[] palette);
 
                 var image = new Image<TPixel>(this.configuration, this.infoHeader.Width, this.infoHeader.Height);
-                using (PixelAccessor<TPixel> pixels = image.Lock())
+
+                Buffer2D<TPixel> pixels = image.Frames.RootFrame.PixelBuffer;
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs]: You've defined `GetRootFramePixelBuffer()` below but are not using it here.

File: src/ImageSharp/ImageFrame{TPixel}.cs [hunk 3]
```diff
@@ -216,19 +216,10 @@ internal ImageFrame(Configuration configuration, ImageFrame<TPixel> source)
         /// It is imperative that the accessor is correctly disposed off after use.
         /// </remarks>
         /// </summary>
-        /// <returns>The <see cref="PixelAccessor{TPixel}"/></returns>
-        internal PixelAccessor<TPixel> Lock()
+        /// <returns>The <see cref="Buffer2D{TPixel}"/></returns>
+        internal Buffer2D<TPixel> Lock()
```
Comment (JimBobSquarePants) [inline src/ImageSharp/ImageFrame{TPixel}.cs]: `Lock` is an odd name now that we are not pinning. Wouldn't `GetRootFramePixelBuffer()` also call this? Is it required anymore?

Comment (antonfirsov) [reply src/ImageSharp/ImageFrame{TPixel}.cs]: Forgot to remove this, good catch!

File: src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs (hunk 1 above)
Comment (antonfirsov) [reply src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs]: Always false of course :)

`65535 * 65535 = 4 294 836 225` and `int.MaxValue = 2 147 483 647` so `Width` and `Height` can't exceed that value in neither case.

`Width` and `Height` should be defined as `long` if we want to allow those sizes. (But I think we don't.)

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs]: I think we're crossing wires here. `65535` is the max value for a `ushort` so not anywhere close to `int.MaxValue`. Incidentally that's the maximum length of any dimensions for a jpeg also.

Comment (antonfirsov) [reply src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs]: Omg true that, I should not make statements before lunch!

Maybe we should define something like an `Image.MaximumDimension` constant.

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs]: I have thought about that before but it varies from format to format so we can't. 

I should print this out and frame it btw, I think it's the first time I've ever corrected you! 😝

Comment (antonfirsov) [reply src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs]: I think it already happened a few times! 😄
Fixed this stuff in my newest commit.

**Comment by JimBobSquarePants** [comment]
All good now, @antonfirsov I'll leave you to merge. 👍

**Pull request merged**